relayer/src/chain/cosmos.rs

```diff
@@ -144,125 +163,136 @@ impl CosmosSdkChain {
         self.rt.block_on(f)
     }
 
-    fn send_tx(&self, proto_msgs: Vec<Any>) -> Result<Vec<IbcEvent>, Error> {
+    fn send_tx(&mut self, proto_msgs: Vec<Any>) -> Result<Response, Error> {
         crate::time!("send_tx");
 
-        let key = self
-            .keybase()
-            .get_key(&self.config.key_name)
-            .map_err(|e| Kind::KeyBase.context(e))?;
-
-        // Create TxBody
-        let body = TxBody {
-            messages: proto_msgs.to_vec(),
-            memo: "".to_string(),
-            timeout_height: 0_u64,
-            extension_options: Vec::<Any>::new(),
-            non_critical_extension_options: Vec::<Any>::new(),
-        };
-
-        // A protobuf serialization of a TxBody
-        let mut body_buf = Vec::new();
-        prost::Message::encode(&body, &mut body_buf).unwrap();
-
-        let mut pk_buf = Vec::new();
-        prost::Message::encode(&key.public_key.public_key.to_bytes(), &mut pk_buf).unwrap();
-
-        crate::time!("PK {:?}", hex::encode(key.public_key.public_key.to_bytes()));
+        let acct_response = self.account()?;
+        if self.acct_seq == 0 {
+            debug!(
+                "send_tx: retrieved account nonce {} for {}",
+                acct_response.sequence,
+                self.id()
+            );
 
-        // Create a MsgSend proto Any message
-        let pk_any = Any {
-            type_url: "/cosmos.crypto.secp256k1.PubKey".to_string(),
-            value: pk_buf,
+            self.acct_seq = acct_response.sequence;
```

Comment:
It looks like we're retrieving here the account information (using `self.account()?;` on line 169) and we use the account sequence number only if this number is not already cached (it gets cached in field `self.acct_seq`). If the sequence is already cached, we ignore the retrieved value. On line 256 we manually increase the cached sequence number (`self.acct_seq += 1;`). Wouldn't it make more sense to _always use_ the retrieved sequence number?